### ISSUE TITLE
MSGFPlusAdapter cannot use configuration files in TOPPAS

### ISSUE THREAD

**Issue by wangm23456** [OpenMS/OpenMS#5499]
in [MSGFPlusAdapter](https://abibuilder.informatik.uni-tuebingen.de/archive/openms/Documentation/release/latest/html/TOPP_MSGFPlusAdapter.html):
> Parameters specified in the configuration file are ignored by MS-GF+ if they are also specified on the command line. 

However, in TOPPAS, there is no way to use only configuration files without using parameter editor parameters.

**Comment by jpfeuffer** [comment]
Hi, same for the command line, where the defaults will be used and overwrite the conf file settings.
That is why the documentation says in the very next sentence, that one can only use the configuration file for "additional" parameters not covered by command line arguments.

**Comment by wangm23456** [comment]
@jpfeuffer This way is very suitable for debugging, but if the software changes its parameters due to version updates, will it cause software errors?

**Comment by jpfeuffer** [comment]
With software you mean MSGFPlus? Yes, if they change an advanced parameter and you did not specify a config file, you might have different results but most likely not an error.
In that case you have to specify all parameters on the command line and all the rest in the config file.
If MSGFPlus changes a parameter (underneath) that we are overwriting with a default, you have to manually change that.
If OpenMS changes defaults, you should be safe, if you save the workflow with all its parameters.
It is very hard to cover all cases but parameters changes are very rare.
If you want full reproducibility, just containerize your versions.

**Comment by wangm23456** [comment]
thanks,
There is also a problem, how do I change the name of the output file? Now I get java.idXML. I want a outputfile named with mass spectrum file.

**Comment by jpfeuffer** [comment]
Yes this is because of a heuristic to choose the output name based on input names.
In this case it somehow picks the name of the java executable input.
@cbielow can you comment on if that can be changed?

**Comment by tapaswenipathak** [comment]
@greengypsy: can you check if there is existing documentation for this? If not, add and close.
